tools/cargo-compare/src/check.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) The cargo-guppy Contributors
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::common::GuppyCargoCommon;
+use crate::GlobalContext;
+use anyhow::{Error, Result};
+use diffus::{edit, Diffable};
+use guppy::PackageId;
+use guppy_cmdlib::CargoMetadataOptions;
+use proptest::prelude::*;
+use proptest::test_runner::{Config, TestCaseResult, TestError, TestRunner};
+use std::collections::{BTreeMap, BTreeSet};
+use std::sync::atomic::{AtomicUsize, Ordering};
+use structopt::StructOpt;
+
+/// Options for cargo/guppy comparisons.
+#[derive(Debug, StructOpt)]
+pub struct CheckOpts {
+    /// Number of randomly generated diff operations to run
+    #[structopt(long)]
+    pub cases: u32,
+    #[structopt(flatten)]
+    pub metadata: CargoMetadataOptions,
+}
+
+impl CheckOpts {
+    /// Executes this command.
+    pub fn exec(self, ctx: &GlobalContext) -> Result<()> {
+        let strat = GuppyCargoCommon::strategy(&self.metadata, ctx.graph);
+
+        let mut testrunner = TestRunner::new(Config {
```

Comment:
Could you set the [`failure_persistence`](https://docs.rs/proptest/0.10.0/proptest/test_runner/struct.Config.html#structfield.failure_persistence) in the config to [`MapFailurePersistence`](https://docs.rs/proptest/0.10.0/proptest/test_runner/struct.MapFailurePersistence.html)? Then you can print out the failing seed and it can later be rerun with the seed. (Bonus if you add the functionality to rerun with the given seed!)